worldedit-core/src/main/java/com/sk89q/worldedit/LocalSession.java

```diff
@@ -1120,6 +1120,24 @@ public Tool getTool(ItemType item, Player player) {
         //FAWE end
     }
 
+    /**
+     * Get the tool assigned to the item.
+     *
+     * @param item the item type
+     * @return the tool, which may be {@code null}
+     */
+    @Nullable
+    @Deprecated
+    //FAWE start
+    //This method is here for byte code compatibility with WorldEdit.
```

Comment:
If that is an upstream method, the FAWE annotations are not needed. Yet, the deprecation note can point to the recommended Fawe method to use.

Reply:
I thought they are needed since the method body differs from WorldEdit. World Edit uses a simple HashMap to store the tools.

I will add a deprecation comment to the method.

Reply:
> I thought they are needed since the method body differs from WorldEdit. World Edit uses a simple HashMap to store the tools.

That would then need to be disclosed within annotations as a one liner

Reply:
I adjusted it and moved the annotations to the body.

It is not a one line since WorldEdit does not use a synchronized block. Should be fine now.